app/services/qa/linked_data/deep_sort_service.rb

```diff
@@ -0,0 +1,265 @@
+# Provide service for for sorting an array of hash based on the values at a specified key in the hash.
+module Qa
+  module LinkedData
+    class DeepSortService
+      attr_reader :a_element, :b_element, :the_array, :sort_key, :preferred_language
+      private :a_element, :b_element, :the_array, :sort_key, :preferred_language
+
+      # @params [Array<Hash<Symbol,Array<RDF::Literal>>>] the array of hashes to sort
+      # @params [sort_key] the key in the hash on whose value the array will be sorted
+      # @returns instance of this class
+      # @example the_array parameter
+      #   [
+      #     {:uri=>[#<RDF::URI:0x3fcff54a829c URI:http://id.loc.gov/authorities/names/n2010043281>],
+      #      :id=>[#<RDF::Literal:0x3fcff4a367b4("n 2010043281")>],
+      #      :label=>[#<RDF::Literal:0x3fcff54a9a98("Valli, Sabrina"@en)>],
+      #      :altlabel=>[],
+      #      :sort=>[#<RDF::Literal:0x3fcff54b4c18("2")>]}
+      #     {:uri=>[#<RDF::URI:0x3fcff54a829c URI:http://id.loc.gov/authorities/names/n201002344>],
+      #      :id=>[#<RDF::Literal:0x3fcff4a367b4("n 201002344")>],
+      #      :label=>[#<RDF::Literal:0x3fcff54a9a98("Cornell, Joseph"@en)>],
+      #      :altlabel=>[],
+      #      :sort=>[#<RDF::Literal:0x3fcff54b4c18("1")>]}
+      #   ]
+      def initialize(the_array, sort_key, preferred_language = :en)
+        @the_array = the_array
+        @sort_key = sort_key
+        @preferred_language = preferred_language
+      end
+
+      # Sort an array of hash on the specified sort key.  The value in the hash at sort key is expected to be an array
+      # with one or more values that are RDF::Literals that translate to a number (e.g. 2), a string number (e.g. "3"),
+      # a string (e.g. "hello"), or a language qualified string (e.g. "hello"@en).
+      # The sort occurs in the following precedence.
+      # * preference for numeric sort (if only one value each and both are integers or a string that can be converted to an integer)
+      # * single value sort (if only one value each and at least one is not an integer)
+      # * multiple values sort (if either has multiple values)
+      # @returns the sorted array
+      def sort
+        return @sorted_array if @sorted_array.present?
+        presort_sortkey_values
+        @sorted_array = @the_array.sort do |a, b|
+          setup_comparison(a, b, sort_key)
+          next numeric_comparator if both_integers?
+          next single_value_comparator if both_single?
+          multiple_value_comparator
+        end
+      end
+
+      private
+
+        def presort_sortkey_values
+          the_array.map! do |element|
+            element[sort_key] = Qa::LinkedData::LanguageSortService.new(element[sort_key], preferred_language).sort
```

Comment:
If the array was mapped to DeepSortElements you could take advantage of sorted_literals in that class here?